File: codes/go/chapter05/answer_A33.go
package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)
	A := make([]int, N)
	for i := 0; i < N; i++ {
		fmt.Scan(&A[i])
	}

	// 全部 XOR した値（ニム和）を求める
	XOR_Sum := A[0]
	for i := 1; i < N; i++ {
		XOR_Sum ^= A[i]
	}

	// 出力
	if XOR_Sum >= 1 {
		fmt.Println("First")
	} else {
		fmt.Println("Second")
	}
}

File: codes/ruby/1/chapter05/answer_A33.rb
# Input
n = gets.to_i
a = gets.split.map(&:to_i)

# Calculate XOR sum
xor_sum = a[0]
for i in 1...n
  xor_sum ^= a[i]
end

# Output
if xor_sum >= 1
  puts "First"
else
  puts "Second"
end
